### PR TITLE
Correct_time()

### LINKED ISSUE #105: Correct datetime of a deployment
From mail discussion about correcting specific datetimes

Qusetion of a user, on Jan 5, 2023: 

> Some time ago I brought up this issue about trying to correct date-time on one of my CTs. [...]

@MarcusRowcliffe's answer:

> I don’t believe there are functions currently available in camtraptor either for correcting times [...].
> I’ve had a quick go at creating basic functions for these tasks (with a warning that there is no error checking so very much up to you ensure sensible input). Your solution to time correction is right I think, but maybe not so easy to reuse! Here’s a suggestion for a more re-usable function:

```r
correct_time <- function(package, deploymentID, wrongTime, rightTime){
  tdiff <- as.POSIXct(rightTime, tz="UTC") - as.POSIXct(wrongTime, tz="UTC")
  i <- package$data$deployments$deploymentID == deploymentID
  package$data$deployments <- dplyr::mutate(package$data$deployments,
                                            start = start + tdiff,
                                            end = end + tdiff)
  package$data$observations <- dplyr::mutate(package$data$observations,
                                             timestamp = timestamp + tdiff)
  package$data$media <- dplyr::mutate(package$data$media,
                                      timestamp = timestamp + tdiff)
  package
}

Example usage:
```r
cpkg <- correct_time(package,
                     deploymentID = "022406d5-96a8-4192-a8d2-f86eab55df57",
                     wrongTime = "2017-01-01 00:00:00",
                     rightTime = "2018-09-01 11:04:00")
```

Thanks @MarcusRowcliffe to put me in cc of this conversation!

### PR DESCRIPTION
**Pull request by sannegovaert**
fix #105

### CODE REVIEW

File: tests/testthat/test-correct_time.R [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+  # Tests
+  expect_identical(
+    deploymentStart_Duration,
+    deploymentStart + duration_Duration
+    )
+  expect_identical(
+    deploymentStart_Duration,
+    as.POSIXct("2020-05-30 04:57:37", tz = "UTC")
+    )
+})
```
Comment (damianooldoni) [inline tests/testthat/test-correct_time.R:139]: We should also test that observations and media are changed.

File: tests/testthat/test-correct_time.R [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  # Tests
+  expect_identical(
+    deploymentStart_difftime,
+    deploymentStart + duration_difftime1
+  )
+  expect_identical(
+    deploymentStart_difftime,
+    as.POSIXct("2020-05-30 04:57:37", tz = "UTC")
+  )
+})
```
Comment (damianooldoni) [inline tests/testthat/test-correct_time.R:187]: Same here: media and observations are also changed and so tests should be added.

**Review by damianooldoni — CHANGES_REQUESTED** [review]
Very nice! And I like the tests a lot too. 

I requested changes as tests should also take into account the changes in `x$data$observations` and `x$data$media`.

I have also committed directly to improve some aspects:
- Remove the "hack" by using the `{{ }}` embracing operator from rlang. See related commit for more information about it.
- Capitalize comments.
- Add some informative comments 
- Remove typos
- Use `purrr::pluck()` instead of `deployments(x)$` syntax as we do in other functions.
- Improve function documentation by adding hyperlinks where needed.
- Use `cli::cli_alert_succes())` for output message. It's visually appealing and the message can be captured as a class.
- Suppress messages where testing are not returning errors: this helps to have a cleaner test console output.
- Solve the test about the returned message.
- Run `devtools::document()`: it modified the rendered documentation of other functions as well.

**Review by damianooldoni — APPROVED** [review]
Well done 👍

**Review by peterdesmet — APPROVED** [review]
Nice work @sannegovaert ! I changed some things, mainly in f6142d66d3617bcd7e2c3c72ac4eb6db47c8b39b

- Indicate in documentation what values get changed
- Don't show all valid deploymentIDs in error message (too verbose)
- Don't warn on duplicates (just do it)
- Only set timestampIssues to FALSE for selected deployments (not all)
- Wrap text in alert_success
- Remove test "correct_time() returns no error on valid deploymentID"

For some reason, I'm getting errors when using `cli::cli_alert_success()` over `cli::cli_inform()`, so I reverted to the latter. I'll create an issue if anyone wants to investigate this further.

**Comment by peterdesmet** [comment]
I have also updated the function name to `shift_time()`, so that users don't get the impression the function can correct the time for them.

**Pull request merged**